timely_beliefs/beliefs/classes.py

```diff
@@ -169,12 +196,42 @@ class BeliefsDataFrame(DataFrame):
     :param: beliefs: a list of TimedBelief objects used to initialise the BeliefsDataFrame
     """
 
-    _metadata = ["sensor"]
+    _metadata = ["sensor", "_event_resolution"]
```

Comment:
Short answer: because the BeliefsDataFrame can contain beliefs from multiple sources, which is indicated by a dedicated index level.

Long answer: we now have a BeliefsDataFrame defined with a unique sensor, but multiple sources, but we could just as well define a BeliefsDataFrame for a unique source, but multiple sensors (and have sensor_id as a dedicated index level). Or even add another index level so it can contain both multiple sensors and multiple sources. This may be a great extension, but also requires some architectural planning (some combinations of sensors don't make sense in terms of what the resulting BeliefsDataFrame is about, e.g. temperature and market price). Still, we might already choose to add a metadata attribute (like we now have for the sensor) whenever we slice the BeliefsDataFrame to select a single sensor, or a single source, or a single belief time, or a single cumulative probability. But we would have to discuss the use case for this.